### PR TITLE
[ONNX] Update peephole pass for prim::ListUnpack

### LINKED ISSUE #45816: ONNX export : unrecognised node type
## 🐛 Bug

I get the following error
```
RuntimeError: !node->kind().is_aten() && !node->kind().is_prim() && !node->kind().is_attr() INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/serialization/export.cpp":453, please report a bug to PyTorch. 
```

I also get these warnings:
```
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
Warning: ONNX Scalar Type Analysis - Scalar types mismatch for tensor inputs of operator onnx::Div. Please report a bug to PyTorch. The scalar type Double of the first tensor is chosen.
```

## To Reproduce

Run this code during an ONNX export. This is from a pytorch port of yolov3.

```
@torch.jit.script
def yolo_predict(pred, anchors, stride: int, scale_x_y: float):
    #transpose
    nA = anchors.size(0)
    nB, nC, nH, nW = pred.size()
    nattr = int(nC / nA)
    pred = pred.view(nB, nA, nattr, nH, nW).permute(0, 1, 3, 4, 2).contiguous()

    #make grid
    grid_x = torch.arange(nW).view(1, 1, 1, nW)
    grid_y = torch.arange(nH).view(1, 1, nH, 1)
    anchor_w = anchors[:, 0].view(1, nA, 1, 1).float() / (nW * stride)
    anchor_h = anchors[:, 1].view(1, nA, 1, 1).float() / (nH * stride)

    #apply grid
    x = pred[..., 0]
    y = pred[..., 1]
    w = pred[..., 2]
    h = pred[..., 3]
    p = pred[..., 4:]
    bx = ((torch.sigmoid(x) - 0.5) * scale_x_y + 0.5 + grid_x) / nW
    by = ((torch.sigmoid(y) - 0.5) * scale_x_y + 0.5 + grid_y) / nH
    bw = torch.exp(w) * anchor_w
    bh = torch.exp(h) * anchor_h
    bp = torch.sigmoid(p)
    preds = torch.stack([bx, by, bw, bh], -1)
    preds = torch.cat([preds, bp], -1)
    preds = preds.view(nB, -1, nattr)

    return preds
```

where anchors is a tensor of shape (3,2), nC is 255, stride is 32 and scale_x_y is 1

## Expected behavior

I expect this to work

## Environment

```
Collecting environment information...
PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.4 LTS (x86_64)
GCC version: (Ubuntu 8.4.0-1ubuntu1~18.04) 8.4.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.16.8

Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1660 Ti
Nvidia driver version: 440.33.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/local/cuda-10.2/targets/x86_64-linux/lib/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.19.2
[pip3] pytorch-msssim==0.1
[pip3] segmentation-models-pytorch==0.1.0
[pip3] torch==1.6.0
[pip3] torchvision==0.7.0
[conda] Could not collect
```

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @gmagogsfm

### PR DESCRIPTION
**Pull request by KsenijaS**
Update pass that handles prim::ListUnpack in peephole file, so that it also covers the case when input to the node is of ListType.

Fixes #45816

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 1]
```diff
@@ -637,11 +637,22 @@ static void fuseListConstructListUnpack(Block* b) {
...
     }
-    if (it->kind() == prim::ListUnpack &&
-        it->input()->node()->kind() == prim::ListConstruct) {
+    if (it->kind() == prim::ListUnpack) {
       for (size_t i = 0; i < it->outputs().size(); i++) {
         auto output = it->outputs().at(i);
-        output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        if (it->input()->node()->kind() == prim::ListConstruct) {
+          output->replaceAllUsesWith(it->input()->node()->inputs().at(i));
+        } else if (it->input()->type()->cast<ListType>()) {
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp:645]: Should this be separated into a function on it's own? The input is not ListConstruct, so not matching up to the method name actually.

Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp:645]: Also wondering is this check enough? 
1. will there be list type input that does not work?
2. will there be other type input (tensor? i'm thinking if it's possible through inferred shape from onnx shape inference) that does work?

Comment (KsenijaS) [reply torch/csrc/jit/passes/onnx/peephole.cpp:645]: yes, I was thinking it's maybe better to change the name of the peephole pass instead

Comment (KsenijaS) [reply torch/csrc/jit/passes/onnx/peephole.cpp:645]: I'm not sure why would element type of a list have any impact?
I can write a test for tensor type.

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 2]
```diff
@@ -632,16 +632,35 @@ static void eraseListConstruct(Block* block, int opset_version) {
 // For ops such as meshgrid where output is a list of Tensors
 // (returns prim::ListConstruct), we need to unpack the list
 // before the pass which deletes ListConstruct.
-static void fuseListConstructListUnpack(Block* b) {
+// This pass also covers the case when the input to ListUnpack
+// is int[] comming from some other op than ListConstruct (like Slice or Shape)
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp:636]:
```suggestion
// is int[] coming from some other op than ListConstruct (like Slice or Shape)
```

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -1703,6 +1703,25 @@ def test_groupnorm_noaffine(self):
         x = torch.randn(4, 6, 180, 180)
         self.run_test(model, x)
 
+    def test_listunpack(self):
+        class ListUnpack(torch.jit.ScriptModule):
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1707]:
```suggestion
        class ListUnpackShape(torch.jit.ScriptModule):
```

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -1703,6 +1703,25 @@ def test_groupnorm_noaffine(self):
...
+            @torch.jit.script_method
+            def forward(self, x):
+                a, b = x.shape
+                return x.new_zeros((a, b))
+
+        x = torch.randn(2, 3)
+        self.run_test(ListUnpack(), x)
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1714]:
```suggestion
        self.run_test(ListUnpackShape(), x)
```

**Review by BowenBao — APPROVED** [review]
LG, thanks

**Comment by KsenijaS** [comment]
@bzinodev can you please review? Thanks!

**Review by bzinodev — APPROVED** [review]

**Comment by BowenBao** [comment]
Test failure related, please have a look @KsenijaS

**Pull request closed without merging**